src/cli/version.rs

```diff
@@ -1,7 +1,80 @@
 use crate::env::release;
 use crate::err::Error;
+use clap::Args;
+use std::fmt::Display;
+use surrealdb::engine::any::connect;
 
-pub fn init() -> Result<(), Error> {
-	println!("{}", release());
+const DEFAULT_ENDOINT: &str = "ws://127.0.0.1:8000";
+
+#[derive(Args, Debug)]
+pub struct VersionCommandArguments {
+	#[command(flatten)]
+	conn: VersionCommandDatabaseConnectionArguments,
+}
+
+#[derive(Args, Debug)]
+pub struct VersionCommandDatabaseConnectionArguments {
+	#[arg(help = "Remote database server url from which to fetch version info")]
+	#[arg(short = 'e', long = "endpoint", visible_aliases = ["conn"])]
+	#[arg(value_parser = super::validator::endpoint_valid)]
+	pub(crate) endpoint: Option<String>,
+}
+
+#[derive(Debug)]
+struct VersionInfo {
+	cli_version: String,
+	server_version: Option<String>,
+}
+
+impl Display for VersionInfo {
+	fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+		write!(f, "CLI: {}", self.cli_version)?;
+
+		if let Some(server_version) = &self.server_version {
+			write!(f, "\nServer: {}", server_version)?;
+		}
+
+		Ok(())
+	}
+}
+
+pub async fn init(
+	VersionCommandArguments {
+		conn: VersionCommandDatabaseConnectionArguments {
+			endpoint,
+		},
+	}: VersionCommandArguments,
+) -> Result<(), Error> {
+	// Initialize opentelemetry and logging
+	crate::o11y::builder().with_log_level("error").init();
+	// Get local CLI version
+	let cli_version = release();
+	// Use user-supplied endpoint or default
+	let endpoint_string = endpoint.clone().unwrap_or_else(|| DEFAULT_ENDOINT.to_string());
+	// Attempt to get server version
+	let server_version_result = get_server_version_string(endpoint_string).await;
+	// If endpoint was user-supplied, propagate errors
+	let server_version = match (endpoint, server_version_result) {
+		(Some(_), Err(e)) => return Err(e),
+		(None, Err(_)) => None,
+		(_, Ok(version)) => Some(version),
+	};
+	// Print version info
+	println!(
+		"{}",
+		VersionInfo {
+			cli_version,
+			server_version
+		}
+	);
 	Ok(())
 }
+
+async fn get_server_version_string(endpoint: String) -> Result<String, Error> {
+	// Connect to the database engine
+	let client = connect(endpoint).await?;
+	// Query database version info
+	let server_version = client.version().await?;
+	// Convert version info to formatted string
+	Ok(format!("{server_version}"))
```

Comment:
`format!("{x}")` should be equivalent to the more concise version `x.to_string()`

Reply:
You right.